src/util/type.cpp

```diff
@@ -106,3 +106,52 @@ bool is_constant_or_has_constant_components(
 
   return false;
 }
+
+static std::string as_string(const struct_typet &src)
```

Comment:
I believe this reinforces even more that `std_types.{h,cpp}` is the right place for this.